models/agent_model.go

```diff
@@ -631,6 +635,46 @@ func (s Agent) TemplateDelimiters(svc *Service) *DelimiterPair {
 	return &tdp
 }
 
+// HashPassword func to calculate password hash. Public and overridable for testing purposes.
+var HashPassword = func(password, salt string) string {
+	buf, err := bcrypt.GenerateFromPasswordAndSalt([]byte(password), bcrypt.DefaultCost, []byte(salt))
+	if err != nil {
+		log.Fatal(err, "cannot calculate hash for password")
+	}
+	return string(buf)
+}
+
+const webConfigTemplate = `basic_auth_users:
+    pmm: {{ . }}
+`
+
+// BuildWebConfigFile builds prometheus-compatible basic auth configuration.
+func (s *Agent) BuildWebConfigFile() string {
+	password := s.GetAgentPassword()
+	salt := getPasswordSalt(s)
+
+	hashedPassword := HashPassword(password, salt)
+
+	var configBuffer bytes.Buffer
+	if tmpl, err := template.New("webConfig").Parse(webConfigTemplate); err != nil {
+		log.Fatalln(err)
+	} else if err = tmpl.Execute(&configBuffer, hashedPassword); err != nil {
+		log.Fatalln(err)
+	}
+
+	config := configBuffer.String()
```

Comment:
log.Fatalln(err) contains os.Exit(1) so clean up looks like
```go
	tmpl, err := template.New("webConfig").Parse(webConfigTemplate)

	if err != nil {
		log.Fatalln(err)
	}

	var configBuffer bytes.Buffer
	if err = tmpl.Execute(&configBuffer, hashedPassword); err != nil {
		log.Fatalln(err)
	}

	config := configBuffer.String()
```

But up to you

Reply:
passed error up to correct handler